### PR TITLE
add Tensor.full and Tensor.full_like and reuse them

### PR DESCRIPTION
**Pull request by ubihot**
Added `Tensor.full` and `Tensor.full_like` and reused them in `zeros, ones, zeros_like, ones_like`

TODO
- need to improve the tests so we don't repeat code

### CODE REVIEW

**Comment by wozeparrot** [comment]
I'm wondering if it would be better to have something like `torch.full_like`.

Maybe switch `Tensor.ones` to be `Tensor.full` as well then add Tensor.ones back calling `Tensor.full`

**Comment by ubihot** [comment]
yeah makes sense, gonna work on it

File: tinygrad/tensor.py [hunk 1]
```diff
@@ -122,13 +122,25 @@ def to(self, device:str):
...
-  def ones(*shape, **kwargs): return Tensor([1], **kwargs).reshape([1]*len(shape)).expand(shape).contiguous()
+  def zeros(*shape, **kwargs): return Tensor.full(shape, 0, **kwargs)
 
   @staticmethod
-  def zeros_like(tensor, dtype:Optional[DType]=None, **kwargs): return Tensor.zeros(*tensor.shape, dtype=tensor.dtype if dtype is None else dtype, **kwargs)
+  def ones(*shape, **kwargs): return Tensor.full(shape, 1, **kwargs)
+
+  @staticmethod
+  def full_like(tensor, fill_value, dtype:Optional[DType]=None, **kwargs):
+    return Tensor.full(tensor.shape, fill_value, dtype=tensor.dtype if dtype is None else dtype, **kwargs)
```
Comment (wozeparrot) [inline tinygrad/tensor.py:137]: Don't think you actually need the dtype to be specified here because it should be part of kwargs.

**Comment by wozeparrot** [comment]
make sure to add tests for `full` and `full_like` as well

File: tinygrad/tensor.py (hunk 1 above)
Comment (ubihot) [reply tinygrad/tensor.py:137]: I've tried to remove it and other stuff but it then complains that dtype is passed twice in kwargs

Comment (wozeparrot) [reply tinygrad/tensor.py:137]: Ah I see, yea should be fine then, this matches what torch does.

**Comment by geohot** [comment]
LGTM!

**Pull request merged**